portfolio-project-matching-app/backend/daoUser.js

```diff
@@ -41,6 +41,28 @@ const createNewUserDoc = async (user) => {
     console.log(`Created user document with id: ${newDocRef.id}`);
 }
 
+const createNewUserDocWithId = async (user, id) => {
+    /*
+    DESCRIPTION:    creates new user document in Firebase Firestore Database
+                    based on user object provided. Similar to
+                    INSERT INTO 'users'. Note, only user document is
+                    created, not associations.
+
+    INPUT:          User object populated with data to be added in new
+                    document
+
+    RETURN:         NA
+    */
+    // build object to be added to Firebase as a user document
+    const userObj = {
+        email: user.email,
+        username: user.username,
+        introduction: user.introduction,
+    };
+    // add user document to Firebase
+    const newDocRef = await addNewDocWithId('users', id, userObj);
+}
+
```

Comment:
Any reason why you didn't stick with the generic `createDocWithId`?

Reply:
I think it was partly me being confused with how the dependencies worked. Since dao.js imports daoUsers.js I was thinking I shouldn't import from dao.js. In hindsight I could still import from dao.js without any issues. I'll fix this and push up my changes.

Reply:
`createDocWithId` makes weird errors for me. Skipping it fixed the issue.

Reply:
I just pushed changes that should fix it.

Reply:
Awesome! No worries at all, I was just curious!